test/types/models.test.ts

```diff
@@ -561,3 +561,9 @@ function findWithId() {
   TestModel.find(id);
   TestModel.findOne(id);
 }
+
+function gh12573ModelAny() {
+  const TestModel = model<any>('Test', new Schema({}));
+  const doc = new TestModel();
+  expectType<any>(doc);
```

Comment:
```suggestion
  const doc = new TestModel();
  expectType<any>(doc);
  const { fieldA } = doc;
  expectType<any>(fieldA);
```

add this test case because of the original issue